### PR TITLE
Deleting a missing pod or job will log a warning

### LINKED ISSUE #489: Deleting a pod or job that does not exist throws exception
Throwing an exception where the original behavior was a no-op is a breaking change #474.   This needs to be updated on a major release.   In the stead, this needs to log a warning stating that the pod or job is missing.

### PR DESCRIPTION
**Pull request by cppwfs**
resolves #489

### CODE REVIEW

File: src/main/java/org/springframework/cloud/deployer/spi/kubernetes/KubernetesTaskLauncher.java [hunk 1]
```diff
@@ -390,7 +391,7 @@ private void deleteJob(String id) {
 				.withLabel(SPRING_APP_KEY, id);
 
 		if (jobsToDelete == null || ObjectUtils.isEmpty(jobsToDelete.list().getItems())) {
-			throw new IllegalStateException(String.format("Cannot delete job for task \"%s\" (reason: job does not exist)", id));
```
Comment (onobc) [inline src/main/java/org/springframework/cloud/deployer/spi/kubernetes/KubernetesTaskLauncher.java:393]: You will need to update the tests as well - you can see what I added in the previous commit https://github.com/spring-cloud/spring-cloud-deployer-kubernetes/commit/c89dd43ba5b7210165ed1278caeaaba712d7bb05

Comment (cppwfs) [reply src/main/java/org/springframework/cloud/deployer/spi/kubernetes/KubernetesTaskLauncher.java:393]: Updated.
Thanks for catching that, I looked at the original commit and missed the added tests.

File: src/test/java/org/springframework/cloud/deployer/spi/kubernetes/KubernetesTaskLauncherIntegrationIT.java [hunk 2]
```diff
@@ -1,5 +1,5 @@
 /*
- * Copyright 2016-2021 the original author or authors.
+ * Copyright 2016-2022 the original author or authors.
```
Comment (onobc) [inline src/test/java/org/springframework/cloud/deployer/spi/kubernetes/KubernetesTaskLauncherIntegrationIT.java:2]: Thank you for catching what I missed last commit ;)

File: src/test/java/org/springframework/cloud/deployer/spi/kubernetes/KubernetesTaskLauncherIntegrationIT.java [hunk 3]
```diff
@@ -47,10 +49,12 @@
  * @author Thomas Risberg
  * @author Chris Schaefer
  * @author Chris Bono
+ * @author Glenn Renfro
  */
 @SpringBootTest(classes = {KubernetesAutoConfiguration.class}, properties = {
 		"spring.cloud.deployer.kubernetes.namespace=default"
 })
+@ExtendWith(OutputCaptureExtension.class)
```
Comment (onobc) [inline src/test/java/org/springframework/cloud/deployer/spi/kubernetes/KubernetesTaskLauncherIntegrationIT.java:57]: Ahh yes. I ❤️ me some `CapturedOutput`

**Review by onobc — APPROVED** [review]

**Comment by onobc** [comment]
Closed via https://github.com/spring-cloud/spring-cloud-deployer-kubernetes/commit/5a0c80adc82b9b665a9bfbd385e1ceac109331d1

**Pull request closed without merging**